### PR TITLE
[SELS-TASK] QuizItem and Choice backend

### PR DESCRIPTION
**Pull request by jstephend-sun**
Asana [Task](https://app.asana.com/0/1201478050789792/1201662721719905/f)

expected output:
api backend
/quizzes/:quiz_id/quiz_items
is available

<<< to navigate to other quiz_items
?item=quiz_item_id 
is the query >>> cancelled

fetch all quiz items for 1 request to api to lessen load

screenshots:
image

with query
image

### CODE REVIEW

File: backend/app/Http/Resources/ChoiceResource.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+use Illuminate\Http\Resources\Json\JsonResource;
+
+class ChoiceResource extends JsonResource {
+    public function toArray($request) {
+        $jsonArr = [];
+        foreach ($this->jsonKeyMap as $k => $v) {
+            $jsonArr[$v] = $this->$k;
+        }
+        return $jsonArr;
```
Comment (ejanton-sun) [inline backend/app/Http/Resources/ChoiceResource.php]: space before `return`
```suggestion
        }
        
        return $jsonArr;
```

File: backend/app/Http/Resources/QuizLogResource.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+use Illuminate\Http\Resources\Json\JsonResource;
+
+class QuizLogResource extends JsonResource {
+    public function toArray($request) {
+        $jsonArr = [];
+        foreach ($this->jsonKeyMap as $k => $v) {
+            $jsonArr[$v] = $this->$k;
+        }
+        return $jsonArr;
```
Comment (ejanton-sun) [inline backend/app/Http/Resources/QuizLogResource.php]: new line before return
```suggestion
        }
        
        return $jsonArr;
```

File: backend/app/Models/Choice.php [hunk 3]
```diff
@@ -9,6 +9,13 @@
 class Choice extends Model {
     use HasFactory;
 
+    public $jsonKeyMap = [
+        'id' => 'id',
+
```
Comment (ejanton-sun) [inline backend/app/Models/Choice.php]: remove new line
```suggestion
```

**Review by ejanton-sun — CHANGES_REQUESTED** [review]

File: backend/app/Http/Resources/ChoiceResource.php (hunk 1 above)
Comment (jstephend-sun) [reply backend/app/Http/Resources/ChoiceResource.php]: done

File: backend/app/Http/Resources/QuizLogResource.php (hunk 2 above)
Comment (jstephend-sun) [reply backend/app/Http/Resources/QuizLogResource.php]: done

File: backend/app/Models/Choice.php (hunk 3 above)
Comment (jstephend-sun) [reply backend/app/Models/Choice.php]: done

**Review by ejanton-sun — APPROVED** [review]

**Pull request merged**